Languages/IronPython/IronPython.Modules/array.cs

```diff
@@ -876,7 +876,7 @@ private class ArrayData<T> : ArrayData {
                     if (!(value is T)) {
                         object newVal;
                         if (!Converter.TryConvert(value, typeof(T), out newVal)) {
-                            if (value != null && typeof(T).IsPrimitive && typeof(T) != typeof(char))
+                            if (value != null && typeof(T).GetTypeInfo().IsPrimitive && typeof(T) != typeof(char))
```

Comment:
I've got an extension method for platforms which don't support `GetTypeInfo` that just returns the `Type` so it works.